nilearn/image/tests/test_image.py

```diff
@@ -837,6 +849,59 @@ def test_math_img(
         assert result.shape == expected_result.shape
 
 
+def test_math_img_copy_default_header(
+    img_4d_ones_eye_default_header, img_4d_ones_eye_tr2
+):
+    # case where data values are not changed and header values are not copied
+    # the result should have default header values
+    formula_no_change = "img * 1"
+    # using img_4d_ones_eye_tr2 with edited header in the formula
+    result = math_img(
+        formula_no_change, img=img_4d_ones_eye_tr2, copy_header_from=None
+    )
+    # header values should instead match default header values
+    assert result.header.__eq__(img_4d_ones_eye_default_header.header)
```

Comment:
```suggestion
    assert result.header == img_4d_ones_eye_default_header.header
```

I believe this should work, no?

Reply:
Just checked and it does indeed work. 
I think I read somewhere that the `__eq__` method is supposed to be more thorough. 
But let's stick to simpler syntax for now.